src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1981,8 +1981,13 @@ def get_multisite_status(self):
             is_multisite_configured = False
         return is_multisite_configured
 
-    def get_multisite_sync_status(self):
+    def get_multisite_sync_status(self, daemon_name: str):
         rgw_multisite_sync_status_cmd = ['sync', 'status']
+        daemons = _get_daemons()
+        if daemon_name:
+            realm_name = daemons[daemon_name].realm_name
+            if realm_name:
+                rgw_multisite_sync_status_cmd.extend(['--rgw-realm', realm_name])
```

Comment:
```suggestion
        daemons = _get_daemons()
        try:
            realm_name = daemons[daemon_name].realm_name
            rgw_multisite_sync_status_cmd.extend(['--rgw-realm', realm_name])
        except (KeyError, AttributeError):
            raise DashboarException....
```

Refactored following principle: ask for forgivness not permission.

No need to check for each variable before using.
Single place (in the catched exception) to deal with any potential failure by for example raising a DashboardException.

This would not be doable with the `get` function unless a default value would be given since in case of failure when looking through the key `realm_name` would be `None`

Reply:
Let me know what you think @afreen23

Reply:
@Pegonzal , am not sure if this is feasible here. If the realm_name is a blank string, it would add --rgw-realm param with a blank string and its not desired here. Moreover the existing approach handles errors through logical checks rather than catching exceptions (KeyError, AttributeError) and this reduces the chances of hidden errors being masked by broad exception handling, which can make debugging harder.

Reply:
Well , I agree we must try to adopt the community adopted practice and pythonic way is more tend towards EAFP rather that LBYL ,  both have there nuances.

EAFP results in cleaner code but  can be slower than LBYL.
As a general rule try using LBYL for operations that are likely to fail and EAFP for operations that are unlikely to fail , so we can have sweet spot for performance maintained too.

As @aaSharma14 mentioned here, we its unlikely to fail "daemons" or "realm_name", so lets follow what @Pegonzal suggested.

For the running of command may be we can do something like:

```
try:
    realm_name = daemons[daemon_name].realm_name
    if  not realm_name: 
       return 
    rgw_multisite_sync_status_cmd.extend(['--rgw-realm', realm_name])
except (KeyError, AttributeError):
    raise DashboarException("Realm name not found for the given daemon.")
```

```
try:
    realm_name = daemons[daemon_name].realm_name
    if  not realm_name:  
               raise DashboarException("Realm name is missing or empty for the given daemon.")
    rgw_multisite_sync_status_cmd.extend(['--rgw-realm', realm_name])
except (KeyError, AttributeError):
    raise DashboarException("Realm name not found for the given daemon.")
```

Please feel free to modify and include better wordings.
Thanks

Reply:
Sure, that's fine @aaSharma14 . Even though if the `realm_name` is a blank string the issue is probably somewhere else which permitted it to be a blank string, not with this code. Or maybe a blank string is an actual possible value and that's why it was permitted.

Anyways, I wrote this as an example of [EAFP](https://docs.python.org/3/glossary.html#term-EAFP) (easier to ask for forgiveness than permission) which is the principle we should try to follow when possible, which we already do in the Dashboard.

But the main takeaway here for this particulare code, as Afreen was mentioning, is that we should catch the `KeyError` exception and proceed accordingly

Reply:
@afreen23 @Pegonzal , If `realm_name` is a blank string, its still a valid scenario because the daemon could be running in "default" zg/zone without a realm. If realm_name is blank we need not pass --rgw-realm param. Anyways I have updated the PR to catch KeyError, Attribute Error. PTAL

Reply:
> EAFP results in cleaner code but can be slower than LBYL.
As a general rule try using LBYL for operations that are likely to fail and EAFP for operations that are unlikely to fail , so we can have sweet spot for performance maintained too.

I agree with you @afreen23. EAFP _can_ be slower if code is likely to fail as catching exceptions is expensive, otherwise it is faster.
I also like your point of going with LBYL for operations that are likely to fail, but that should be the exception in my opinion.

What I disagree with is focusing in the performance. Mainly due to the Pareto rule which applied to the performance of a system states that by tackling the 20% of critical bottlenecks will improve the software by 80%. 
This descourages from making premature optimization changes that lack evidence which will result in little improvement and will damage clean code.

Anyways, this is far beyond this PR's scope, so I don't wanan bother Aashish more here, but definitely a great topic to discuss about, I'll try to bring some of this topics in the Wiki, so maybe we can try to set on a clear pattern/practices for the future

Thanks @afreen23, @aaSharma14  for your thoughts and discussion on this

Reply:
Agree @Pegonzal premature optimization is evil.
We should review such practices as per use cases, 👍🏿